src/main/java/dataStructures/hashSet/openAddressing/HashSet.java

```diff
@@ -121,15 +134,29 @@ public boolean remove(T element) {
      * @return true if this set contains the specified element
      */
     public boolean contains(T element) {
-        int bucketIndex = this.search(element);
+        int collisions = 0;
+        while (collisions < capacity()) {
+            int bucketIndex = hashFunction(element, collisions);
+
+            // Invariant: Probe sequence is unbroken (no null values between buckets in the sequence).
+            // This is maintained by add and delete.
+            // This means that given a probe sequence e.g. (1, 2, 3, 4, 5, ...) for a given element, add will attempt to
+            // add the element into the buckets in the given order. As a result, if an element is in bucket 3, there
+            // will be elements in buckets 1 and 2, given that there must have been collisions for the element to be
+            // added to bucket 3 instead of bucket 1, or bucket 2.
+            // Similarly, to maintain that invariant, delete will not replace the element with null, but with a
+            // marker (Tombstone).
+            if (isNullBucket(bucketIndex)) {
+                return false;
+            }
 
-        if (bucketIndex == ELEMENT_NOT_FOUND) {
-            return false;
+            if (buckets[bucketIndex].equals(element)) {
```

Comment:
does having this line of code here process TombStones correctly? Will Tombstone evaluate to false here and move on? as suggested in line 156?

Reply:
Added tests in 123c8cc1dfea21cef8e3a5636c9b16a20f9da92d